tinydb/table.py

```diff
@@ -230,20 +231,27 @@ def all(self) -> List[Document]:
         # of all documents by using the ``list`` constructor to perform the
         # conversion.
 
+        if limit > 0:
+            return list(iter(self))[:limit]
```

Comment:
The `list(iter(self))` bit will store all elements of `self` in memory to throw possibly most of them away when slicing with `[:limit]`.
I suggest "slicing" the _iterator itself_ instead by making use of [`itertools.islice`](https://docs.python.org/3/library/itertools.html#itertools.islice):
```suggestion
            return list(itertools.islice(iter(self), limit))
```

Reply:
Thank you.